invoices/invoiceregistry_test.go

```diff
@@ -547,3 +550,84 @@ func TestUnknownInvoice(t *testing.T) {
 		t.Fatal("expected invoice not found error")
 	}
 }
+
+// Tests that invoices are canceled after expiration.
+func TestInvoiceExpiryWithRegistry(t *testing.T) {
+	t.Parallel()
+
+	cdb, cleanup, err := newTestChannelDB()
+	defer cleanup()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expiryWatcher := NewInvoiceExpiryWatcher(&testClock{})
+	registry := NewRegistry(cdb, expiryWatcher, testFinalCltvRejectDelta)
+
+	// First prefill the Channel DB with some pre-existing invoices,
+	// half of them still pending, half of them expired.
+	const numExpired = 50
+	const numPending = 50
```

Comment:
Does it add something to use that many test invoices?

Reply:
Changed it. PTAL